tests/refc/strings/TestStrings.idr

```diff
@@ -0,0 +1,43 @@
+module TestStrings
+
+import Data.String
+import Data.String.Iterator
+
+iteratorTail : String -> String
+iteratorTail str = withString str $ \it => unconsTail str (uncons str it)
+  where
+    unconsTail : (str : String) -> (1 _ : UnconsResult str) -> String
+    unconsTail str EOF = ""
+    unconsTail str (Character _ tailIt) = withIteratorString str tailIt id
+
+main : IO ()
+main = do
+    let helloWorld = "Hello, " ++ "world"
+
+    putStrLn helloWorld
+    putStrLn $ show $ length helloWorld
+
+    putStrLn $ reverse helloWorld
+    putStrLn $ substr 1 2 helloWorld
+    putStrLn $ show $ assert_total $ strIndex helloWorld 1
+
+    putStrLn $ strCons 'a' "bc"
+    putStrLn $ show $ strUncons "abc"
+
+    putStrLn $ fastPack ['p', 'a', 'c', 'k']
```

Comment:
Can we get some `fastPack` & so on tests on empty strings too? Last time around we had a bug
precisely when the input string was empty.

Reply:
Good call. Upon closer inspection I made an off-by-one error in `fastConcat`.